project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/execution/gradle/GradleBuildExceptionProvider.kt

```diff
@@ -81,77 +158,49 @@ class GradleBuildExceptionProvider : BuildExceptionProvider {
-                )
-            }
-            processAdapter.getRunResult()  // Wait until writeAction is done
-        } finally {
-            Disposer.dispose(endProcessDisposable)
-        }
-    }
+        logger.info { "Run gradle task: ${task.executableName}" }
 
-    private fun Raise<CompilationPropertyCheckerError>.buildConfiguration(
-        project: Project,
-        task: ExecutableGradleTask,
-    ): GradleRunConfiguration {
-        val configurationFactory = GradleExternalTaskConfigurationType.getInstance().factory
-        val configuration = GradleRunConfiguration(project, configurationFactory, "Gradle Test Project Compilation")
-        configuration.settings.apply {
-            externalSystemIdString = GradleConstants.SYSTEM_ID.id
-            taskNames = listOf(task.executableName)
-            externalProjectPath = project.guessProjectDir()?.path
-                ?: raise(CompilationPropertyCheckerError.InvalidBuildSystem)
-            isPassParentEnvs = true
-            scriptParameters = "--quiet --no-configuration-cache"
-        }
-        return configuration
-    }
+        val std = ByteArrayOutputStream()
+        val err = ByteArrayOutputStream()
 
-    private fun buildEnvironment(
-        project: Project,
-        executor: Executor,
-        configuration: GradleRunConfiguration,
-        processAdapter: GradleRunProcessAdapter,
-        endProcessDisposable: Disposable,
-    ): ExecutionEnvironment =
-        ExecutionEnvironmentBuilder
-            .create(executor, configuration)
-            .target(ExecutionTargetManager.getActiveTarget(project))
-            .build { descriptor ->
-                descriptor.processHandler?.let {
-                    it.addProcessListener(processAdapter, endProcessDisposable)
-                    Disposer.register(endProcessDisposable) {
-                        if (!it.isProcessTerminated && !it.isProcessTerminating) {
-                            it.destroyProcess()
-                        }
+        val exitCode = executeWithGradleConnection(context) { connection, cont, javaHome ->
+            connection.newBuild()
+                .forTasks(task.executableName)
+                .setStandardOutput(std)
+                .setStandardError(err)
+                .setJavaHome(javaHome?.let { File(it) })
+                .withArguments("--no-configuration-cache", "--quiet")
+                .run(object : ResultHandler<Void> {
+                    override fun onComplete(result: Void?) {
+                        cont.resume(EXIT_CODE_OK)
                     }
-                }
-            }
+
+                    override fun onFailure(exception: GradleConnectionException) {
+                        cont.resume(EXIT_CODE_FAIL)
+                    }
+                })
+        }
+
+        val error = err.toString(Charsets.UTF_8)
+        val stdout = std.toString(Charsets.UTF_8)
+        logger.debug { "STD: $stdout" }
+        logger.debug { "ERR: $error" }
```

Comment:
I think it better to put it to trace, otherwise we will have a lot of duplicated output